grails-app/services/org/pih/warehouse/invoice/PrepaymentInvoiceMigrationService.groovy

```diff
@@ -0,0 +1,83 @@
+package org.pih.warehouse.invoice
+
+import grails.gorm.transactions.Transactional
+
+import org.pih.warehouse.core.Constants
+import org.pih.warehouse.order.OrderAdjustment
+import org.pih.warehouse.order.OrderItem
+import org.pih.warehouse.order.OrderItemStatusCode
+
+/**
+ * This service exists for the sole purpose of performing data migrations for changelog 0.9.x/2024-08-20-0000.
+ * The logic was only put in a service class so that it could be unit tested. Do not use it in regular flows.
+ */
+@Transactional
+class PrepaymentInvoiceMigrationService {
```

Comment:
I started wondering if it is a good idea to move this logic to a separate service, because we are removing the liquibase's complaints about checksum if someone modifies something in this service

Reply:
Plus I am getting more and more convinced that it should be a manual trigger from `data migration` page in app instead of a db migration ;/

Reply:
that's a fair point, though I'd hope we wouldn't need to ever modify this file and if we did I don't see a world where we'd want to rerun it. The only reason it's in here is so that I could unit test it.

As to making this a manual trigger, yes we discussed it before and that's likely a safer option (that or just auto-migrate any invoices when they're saved if they haven't been migrated yet). We'd need to refactor the partial invoicing code to be able to support old data, but it'd mean we're not running this questionably long migration step on startup.